src/app/components/hero/hero.component.ts

```diff
@@ -27,155 +30,41 @@ import { City } from '../../models/cities.enum';
     provideIcons({ bootstrapSearch, bootstrapBuilding, bootstrapStack }),
   ],
 })
-export class HeroComponent {
+export class HeroComponent implements OnInit {
   technology: WritableSignal<string> = signal('');
-  #allEntreprises: IEntreprise[] = this.getFakeEntrepriseData();
+  #allEntreprises: IEntreprise[];
```

Comment:
```suggestion
  #allEntreprises: Signal<IEntreprise[]> = toSignal(
    this.entrepriseService.getFakeEntrepriseData()
  );
```
you can use toSignal without needing to subscribe or an ngOnInit method

Reply:
@abdelhalimjean With this approach, if we are not going to use Observable, and instead use only signals, should I modify my service methode also because it is returning an observable :  
getFakeEntrepriseData(): Observable<IEntreprise[]> {
    return this.http.get<IEntreprise[]>(this.dataUrl);
  }, and i get this error :Type 'WritableSignal<Observable<IEntreprise[]>>' is not assignable to type 'Signal<IEntreprise[]>'.
  Type 'WritableSignal<Observable<IEntreprise[]>>' is not assignable to type '() => IEntreprise[]'.
    Type 'Observable<IEntreprise[]>' is missing the following properties from type 'IEntreprise[]': length, pop, push, concat, and 34 more.ts(2322

Reply:
you can keep the service as is, but in the component , use `toSignal` as suggested, it's a method that transform an observable to a signal, in thsi example it's transforming `this.entrepriseService.getFakeEntrepriseData()` which is `Observable<IEntreprise[]>` to `Signal<IEntreprise[]>`

https://angular.dev/guide/signals/rxjs-interop#tosignal